libvmi/cache.c

```diff
@@ -162,12 +164,23 @@ pid_cache_set(
     vmi_pid_t pid,
     addr_t dtb)
 {
-    gint *key = (gint *) safe_malloc(sizeof(gint));
+    gint *key = (gint *) g_malloc0(sizeof(gint));
+
+    if ( !key )
+        return;
 
     *key = pid;
     pid_cache_entry_t entry = pid_cache_entry_create(pid, dtb);
 
-    g_hash_table_insert(vmi->pid_cache, key, entry);
+    if ( !entry )
+        return;
+
+    if ( !g_hash_table_insert(vmi->pid_cache, key, entry) )
```

Comment:
The return value of g_hash_table_insert was unchecked deliberately here:  GLIB 2.38 and earlier returned void rather than gboolean 
https://developer.gnome.org/glib/2.38/glib-Hash-Tables.html#g-hash-table-insert

Supporting slightly older platforms is important, e.g. RHEL 6, so we need compat checks for these (or return to old unchecked behavior. tossup..result ends up being the same for a failure with current logic in most locations)

Lines 585 and 603 will generate a errors like "error: invalid use of void expression" and "error: void value not ignored as it ought to be" in their current form on older platforms as well.